packages/web3-common/src/eth_execution_api.ts

```diff
@@ -0,0 +1,293 @@
+	readonly accessList: AccessList;
+}
+
+export interface Transaction1559Signed extends Transaction1559Unsigned {
+	readonly yParity: Uint;
+	readonly r: Uint;
+	readonly s: Uint;
+}
+
+export interface Transaction2930Unsigned extends BaseTransaction {
+	readonly gasPrice: Uint;
+	readonly accessList: AccessList;
+}
+
+export interface Transaction2930Signed extends Transaction2930Unsigned {
+	readonly yParity: Uint;
+	readonly r: Uint;
+	readonly s: Uint;
+}
+
+export interface TransactionLegacyUnsigned extends BaseTransaction {
+	readonly gasPrice: Uint;
+}
+
+export interface TransactionLegacySigned extends TransactionLegacyUnsigned {
+	readonly v: Uint;
+	readonly r: Uint;
+	readonly s: Uint;
+}
+
+// https://github.com/ethereum/execution-apis/blob/main/src/schemas/transaction.json#L178
+export type TransactionUnsigned =
+	| Transaction1559Unsigned
+	| Transaction2930Unsigned
+	| TransactionLegacyUnsigned;
+
+// https://github.com/ethereum/execution-apis/blob/main/src/schemas/transaction.json#L262
+export type TransactionSigned =
+	| Transaction1559Signed
+	| Transaction2930Signed
+	| TransactionLegacySigned;
+
+// https://github.com/ethereum/execution-apis/blob/main/src/schemas/transaction.json#L269
+export type TransactionInfo = TransactionSigned & {
+	readonly blockHash: HexString32Bytes | null;
+	readonly blockNumber: Uint | null;
+	readonly from: Address;
+	readonly hash: HexString32Bytes;
+	readonly transactionIndex: Uint | null;
+};
+
+// https://github.com/ethereum/execution-apis/blob/main/src/schemas/transaction.json#L24
+export type TransactionWithSender = TransactionUnsigned & { from: Address };
+
+// https://github.com/ethereum/execution-apis/blob/main/src/schemas/block.json#L2
+export interface Block {
+	readonly parentHash: HexString32Bytes;
+	readonly sha3Uncles: HexString32Bytes;
+	readonly miner: HexString;
+	readonly stateRoot: HexString32Bytes;
+	readonly transactionsRoot: HexString32Bytes;
+	readonly receiptsRoot: HexString32Bytes;
+	readonly logsBloom: HexString256Bytes | null;
+	readonly difficulty?: Uint;
+	readonly number: Uint | null;
+	readonly gasLimit: Uint;
+	readonly gasUsed: Uint;
+	readonly timestamp: Uint;
+	readonly extraData: HexStringBytes;
+	readonly mixHash: HexString32Bytes;
+	readonly nonce: HexStringBytes | null;
+	readonly totalDifficulty: Uint;
+	readonly baseFeePerGas?: Uint;
+	readonly size: Uint;
+	readonly transactions: TransactionHash[] | TransactionSigned[];
+	readonly uncles: Uncles;
+	readonly hash: HexString32Bytes | null;
+}
+
+// https://github.com/ethereum/execution-apis/blob/main/src/schemas/filter.json#L59
+export type Topic = HexString256Bytes;
+
+// https://github.com/ethereum/execution-apis/blob/main/src/schemas/receipt.json#L2
+export interface Log {
+	readonly removed?: boolean;
+	readonly logIndex?: Uint;
+	readonly transactionIndex?: Uint;
+	readonly transactionHash?: HexString32Bytes;
+	readonly blockHash?: HexString32Bytes;
+	readonly blockNumber?: Uint;
```

Comment:
```suggestion
	readonly logIndex?: Uint | null;
	readonly transactionIndex?: Uint | null;
	readonly transactionHash?: HexString32Bytes | null;
	readonly blockHash?: HexString32Bytes | null;
	readonly blockNumber?: Uint | null;
```

Legacy spec:

![image](https://user-images.githubusercontent.com/20375610/140855545-03ffb273-3330-423c-9195-debc917913a8.png)